worldedit-core/src/main/java/com/sk89q/worldedit/world/snapshot/SnapshotRepository.java

```diff
@@ -92,11 +95,28 @@ public SnapshotRepository(String dir) {
                     detectDate(snapshot);
                     list.add(snapshot);
                 }
-            } else if (file.isDirectory() && file.getName().equalsIgnoreCase(worldName)) {
-                for (String name : file.list(filter)) {
-                    Snapshot snapshot = new Snapshot(this, file.getName() + "/" + name);
-                    detectDate(snapshot);
-                    list.add(snapshot);
+            } else if (file.isDirectory()) {
+                if (file.getName().equalsIgnoreCase(worldName)) {
+                    // <world>/<date>
+                    for (String name : file.list(filter)) {
+                        Snapshot snapshot = new Snapshot(this, file.getName() + "/" + name);
+                        detectDate(snapshot);
+                        list.add(snapshot);
+                    }
+                } else {
+                    Calendar date = fileNameDateParser.detectDate(file);
+                    if (date != null) {
+                        // <date>/<world>
+                        ZonedDateTime zdt = ZonedDateTime.ofInstant(
+                            date.toInstant(),
+                            ZoneOffset.UTC
+                        );
+                        for (String name : file.list((dir, name) -> name.equalsIgnoreCase(worldName))) {
```

Comment:
```suggestion
                        for (String name : file.list((dir, name) -> Files.getNameWithoutExtension(name).equalsIgnoreCase(worldName))) {
```
or else `world.zip` won't match `world`

Reply:
Also `filter` is ignored here so potentially non-snapshot files will be listed if they match the world name.